embeds/ge2017-nugget/_source/main.js

```diff
@@ -1,88 +1,127 @@
-        if (typeof document.body[fn] == 'function') {
-            matchesFn = fn;
-            return true;
+    function closest(el, selector) {
+        var matchesFn;
+
+        // find vendor prefix
+        ['matches','webkitMatchesSelector','mozMatchesSelector','msMatchesSelector','oMatchesSelector'].some(function(fn) {
+            if (typeof document.body[fn] == 'function') {
+                matchesFn = fn;
+                return true;
+            }
+            return false;
+        })
+
+        var parent;
+
+        // traverse parents
+        while (el) {
+            parent = el.parentElement;
+            if (parent && parent[matchesFn](selector)) {
+                return parent;
+            }
+            el = parent;
         }
-        return false;
-    })
 
-    var parent;
+        return null;
+    }
 
-    // traverse parents
-    while (el) {
-        parent = el.parentElement;
-        if (parent && parent[matchesFn](selector)) {
-            return parent;
+    var SUPPORTER_COOKIE = "gu_paying_member";
+
+    var CONTRIBUTION_COOKIE = "gu.contributions.contrib-timestamp";
+
+    // Taken from https://www.w3schools.com/js/js_cookies.asp
+    function getCookie(cname) {
+        var name = cname + "=";
+        var decodedCookie;
+        try {
+            decodedCookie = decodeURIComponent(document.cookie);
+        }
+        catch(err) {
+            return null;
+        }
+        var ca = decodedCookie.split(';');
+        for(var i = 0; i <ca.length; i++) {
+            var c = ca[i];
+            while (c.charAt(0) == ' ') {
+                c = c.substring(1);
+            }
+            if (c.indexOf(name) == 0) {
+                return c.substring(name.length, c.length);
+            }
         }
-        el = parent;
+        return null;
     }
 
-    return null;
-}
+    function isPayingMember() {
+        var cookieValue = getCookie(SUPPORTER_COOKIE)
+        return cookieValue && cookieValue !== "false";
+    }
 
+    function isContributor() {
+        return !!getCookie(CONTRIBUTION_COOKIE)
+    }
 
-setTimeout(function() {
-    // find this thrasher
-    var thrasher = document.querySelector('.ge2017-supporter-nugget');
+    function shouldReaderSeeGoldenNugget() {
+        return !isPayingMember() && !isContributor()
+    }
 
-    // the thrasher's section tag
-    var thrasherContainer = closest(thrasher, 'section');
+    if (shouldReaderSeeGoldenNugget()) {
 
-    // section where the treat should go
-    var targetContainer = findTargetContainer(thrasherContainer);
+        // find this thrasher
+        var thrasher = document.querySelector('.ge2017-supporter-nugget');
 
-    console.log('target:', targetContainer);
+        // the thrasher's section tag
+        var thrasherContainer = closest(thrasher, 'section');
 
+        // section where the treat should go
+        var targetContainer = findTargetContainer(thrasherContainer);
```

Comment:
I think we need a null check here in case `closest(thrasher, 'section')` returns null